examples/src/main/scala/example/akkahttp/AuthExampleApp.scala

```diff
@@ -11,53 +11,83 @@ import caliban.{ AkkaHttpAdapter, RootResolver }
 import sttp.model.StatusCode
 import sttp.tapir.json.circe._
 import sttp.tapir.model.ServerRequest
-import zio.{ FiberRef, RIO, Runtime, Unsafe, ZIO, ZLayer }
+import zio.{ RIO, Runtime, Unsafe, ZIO, ZLayer }
 
 import scala.concurrent.ExecutionContextExecutor
 import scala.io.StdIn
+import zio.Scope
+import _root_.cats.effect.Fiber
 
 object AuthExampleApp extends App {
 
   case class AuthToken(value: String)
 
-  type Auth = FiberRef[Option[AuthToken]]
+  type Auth = AuthToken
 
-  object AuthInterceptor extends RequestInterceptor[Auth] {
-    override def apply[R <: Auth, A](
+  object AuthInterceptor extends RequestInterceptor[Any, Auth] {
+    override def apply[R, A](
       request: ServerRequest
-    )(effect: ZIO[R, TapirResponse, A]): ZIO[R, TapirResponse, A] =
+    )(effect: ZIO[R with Auth, TapirResponse, A]): ZIO[R, TapirResponse, A] =
       request.headers.collectFirst {
         case header if header.is("token") => header.value
       } match {
-        case Some(token) => ZIO.serviceWithZIO[Auth](_.set(Some(AuthToken(token)))) *> effect
+        case Some(token) =>
+          effect.provideSomeLayer[R](ZLayer.succeed[Auth](AuthToken(token)))
```

Comment:
I'm wondering if performance is not going to get worse with that change. `provide` in general has some cost because it's merging environments together. I know recently zio-grpc completely removed the usage of environment to pass context for this reason (which requires many heavy changes for me at work 😭 ).

On the other hand it does look simpler to use since `FiberRef` is no longer necessary. I'm interested in other opinions @paulpdaniels @frekw @kyri-petrou ?

Reply:
Is there a benchmark I can run to see what the impact is? I guess that should actually take the number of interceptors into account 🤔 If needed I can set up a benchmark to get some actual numbers on the impact.

Reply:
I _think_ that this change is okay because the number of interceptors is likely going to be very small (most cases 0 or 1), so it'll be called just once per request. Also, users looking to maximise performance are likely going to implement this without merging context anyways. Since this is just the example, we could add a comment to indicate that users wishing to maximise performance could implement this via a `FiberRef` instead. Or is it no longer possible to implement this using a `FiberRef` given the changes in this PR?

Reply:
I don't have any performance number, but I think using `ZLayer.apply` in ZIO2 is more efficient, and probably what we want in most cases (eliminating a single dep from `R`):

```scala
ZLayer.succeed(AuthToken(token)(effect)
```

Reply:
I am supportive of this change since it's also more inline with how `R` is described for ZIO2:

> The most idiomatic use of the ZIO environment is for:

> - Describing as values workflows that use capabilities that are "local" to a particular context
> - Where tracking the use of this context at the type level is helpful for reasoning about programs

(from https://zio.dev/reference/contextual/zio-environment-use-cases#1-local-contexts)

Reply:
Yeah I think it should be okay performance-wise, and it's quite simpler than with `FiberRef` actually. I think you can remove the `FiberRef` example, if we go for this solution it should be the recommended way.

The only concern I have is complexity as @kyri-petrou mentioned.